Scripts/Runtime/Core/Util/MathExtension.cs

```diff
@@ -0,0 +1,406 @@
+        /// <param name="b">The <see cref="float4x4"/> to compare.</param>
+        /// <returns>
+        /// True (component-wise) if the delta between <see cref="a"/> and <see cref="b"/> is less than
+        /// <see cref="MathUtil.FLOATING_POINT_EQUALITY_TOLERANCE"/>.
+        /// </returns>
+        /// <remarks>
+        /// Resolves infinite and NaN comparisons as true with the same logic as <see cref="IsEqualOrNaN"/> at
+        /// additional cost vs <see cref="IsApproximately"/>.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool4x4 IsApproximatelySafe(this float4x4 a, float4x4 b)
+        {
+            return new bool4x4(
+                a.c0.IsApproximatelySafe(b.c0),
+                a.c1.IsApproximatelySafe(b.c1),
+                a.c2.IsApproximatelySafe(b.c2),
+                a.c3.IsApproximatelySafe(b.c3)
+            );
+        }
+
+        /// <summary>
+        /// Converts the components of a value to infinite quantities where:
+        /// >0 = <see cref="float.PositiveInfinity"/>
+        /// <0 = <see cref="float.NegativeInfinity"/>
+        /// 0 = 0
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <returns>A signed infinite value.</returns>
+        public static float3 ToSignedInfinite(this float3 value)
+        {
+            return math.select(math.sign(value) * new float3(float.PositiveInfinity), value, value == 0);
+        }
+
+        /// <summary>
+        /// Compares two <see cref="float"/>s and returns true if the components are equal.
+        /// Unlike <see cref="float.Equals"/> this method considers <see cref="float.NaN"/> == <see cref="float.NaN"/>
+        /// to be true.
+        /// </summary>
+        /// <param name="a">The <see cref="float"/> to compare.</param>
+        /// <param name="b">The <see cref="float"/> to compare.</param>
+        /// <returns>True if the values are equal or both <see cref="float.NaN"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsEqualOrNaN(this float a, float b)
+        {
+            return a == b | (math.isnan(a) & math.isnan(b));
+        }
+
+        /// <summary>
+        /// Compares two <see cref="float2"/>s and returns true if the components are equal.
+        /// Unlike <see cref="float2.Equals"/> this method considers <see cref="float.NaN"/> == <see cref="float.NaN"/>
+        /// to be true.
+        /// </summary>
+        /// <param name="a">The <see cref="float2"/> to compare.</param>
+        /// <param name="b">The <see cref="float2"/> to compare.</param>
+        /// <returns>True (component-wise) if the values are equal or both <see cref="float.NaN"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool2 IsEqualOrNaN(this float2 a, float2 b)
+        {
+            return a == b | (math.isnan(a) & math.isnan(b));
```

Comment:
Same question above about mixed values + NaN `float2(5.0f, NaN)`

Reply:
I think the same answer as above?

``` csharp
a == b   // (true, false)
(math.isnan(a) & math.isnan(b))   // (false, true)

(true, false) | (false, true)   // (true, true)
```

In this case, I do have a test already :)
https://github.com/decline-cookies/anvil-unity-core/blob/c0d4afce82153b6904e143c5ff6046cda53d1d7a/Scripts/Editor/Tests/Core/Util/MathExtensionTests.cs#L513

Reply:
Yep all good